vault/identity_store_oidc.go

```diff
@@ -509,25 +512,48 @@ func (i *IdentityStore) pathOIDCCreateUpdateKey(ctx context.Context, req *logica
 		return logical.ErrorResponse("unknown signing algorithm %q", key.Algorithm), nil
 	}
 
+	now := time.Now()
+
 	// Update next rotation time if it is unset or now earlier than previously set.
-	nextRotation := time.Now().Add(key.RotationPeriod)
+	nextRotation := now.Add(key.RotationPeriod)
 	if key.NextRotation.IsZero() || nextRotation.Before(key.NextRotation) {
 		key.NextRotation = nextRotation
 	}
 
-	// generate keys if creating a new key or changing algorithms
+	// generate current and next keys if creating a new key or changing algorithms
 	if key.Algorithm != prevAlgorithm {
 		signingKey, err := generateKeys(key.Algorithm)
 		if err != nil {
 			return nil, err
 		}
 
+		i.Logger().Debug("generated OIDC public key to sign JWTs", "key_id", signingKey.Public().KeyID)
 		key.SigningKey = signingKey
-		key.KeyRing = append(key.KeyRing, &expireableKey{KeyID: signingKey.Public().KeyID})
+		key.KeyRing = append(key.KeyRing, &expireableKey{
+			KeyID:    signingKey.Public().KeyID,
+			ExpireAt: now.Add(key.RotationPeriod).Add(key.VerificationTTL),
```

Comment:
IIUC, this means that we're calculating the ExpireAt by factoring the rotation period early on instead of just appending the VerificationTTL upon rotation like it currently is. What happens if there's an update on the named key's `rotation_period` after the key has been created (i.e. by an update)?